tsdb/store_test.go

```diff
@@ -262,6 +262,70 @@ func TestStore_BadShard(t *testing.T) {
 	}
 }
 
+func TestStore_BadShardClear(t *testing.T) {
+	const errStr = "a shard open error"
+	indexes := tsdb.RegisteredIndexes()
+	for _, idx := range indexes {
+		func() {
+			s := MustOpenStore(t, idx)
+			defer require.NoErrorf(t, s.Close(), "closing store with index type: %s", idx)
```

Comment:
Doesn't this close the store immediately, because arguments to deferred functions are evaluated at the `defer` line?  See [here](https://goplay.tools/snippet/1PRjy_Npvqz)

Reply:
Hm interesting - this similar construct is used through out the tests (outside of the ones I worked on) 

I'll go through this and see what's going on.

Reply:
Take a look at the Go playground link I shared for a demonstration.